Objects/dictobject.c

```diff
@@ -2153,6 +2153,86 @@ _PyDict_FromItems(PyObject *const *keys, Py_ssize_t keys_offset,
     return dict;
 }
 
+
+PyObject*
+_PyDict_FromStackRefItems(_PyStackRef const *keys, Py_ssize_t keys_offset,
```

Comment:
Here too I think we should maybe have `_PyDict_FromItems` just call `_PyDict_FromStackRefItems` with the `keys` and `values` casted to `_PyStackRef*`